packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -126,7 +126,7 @@ function specifierNameMatches(type: ts.Type, name: string | string[]): boolean {
   if (symbol === undefined) {
     return false;
   }
-  return name.some(item => item === symbol.escapedName);
+  return name.some(item => (item as ts.__String) === symbol.escapedName);
```

Comment:
I think just keep using a disable comment for these cases and that way when we fix https://github.com/typescript-eslint/typescript-eslint/issues/7114 we'll be able to just remove the unnecessary disables